rest/upgradetest/upgrade_registry_test.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2023-Present Couchbase, Inc.
+
+Use of this software is governed by the Business Source License included in
+the file licenses/BSL-Couchbase.txt.  As of the Change Date specified in that
+file, in accordance with the Business Source License, use of this software will
+be governed by the Apache License, Version 2.0, included in the file
+licenses/APL2.txt.
+*/
+
+package upgradetest
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/couchbase/sync_gateway/base"
+	"github.com/couchbase/sync_gateway/rest"
+	"github.com/stretchr/testify/require"
+)
+
+// TestDefaultMetadataID creates an database using the named collections on the default scope, then modifies that database to use
+// only the default collection. Verifies that metadata documents are still accessible.
+func TestDefaultMetadataIDNamedToDefault(t *testing.T) {
+	base.RequireNumTestDataStores(t, 2)
+	base.SetUpTestLogging(t, base.LevelInfo, base.KeyAll)
+	rtConfig := &rest.RestTesterConfig{
+		CustomTestBucket: base.GetPersistentTestBucket(t),
```

Comment:
I think PersistentConfig should imply `GetPersistentTestBucket` and that should go on `RestTester`. I may have already made this change?